wolfcrypt/test/test.c

```diff
@@ -3141,9 +3147,17 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t sha256_test(void)
     byte      hashcopy[WC_SHA256_DIGEST_SIZE];
     wc_test_ret_t ret = 0;
 
-    testVector a, b, c;
-    testVector test_sha[3];
+    testVector a, b, c, d;
+    testVector test_sha[4];
+#ifndef NO_WOLFSSL_SHA256_INTERLEAVE
```

Comment:
This `NO_WOLFSSL_SHA256_INTERLEAVE` macro is new and only applies to test.c correct? If so the name should include "TEST" in it. Also needs documented.
I believe the interleave avoids the copy, update, final scenario?

Reply:
> This `NO_WOLFSSL_SHA256_INTERLEAVE` macro is new and only applies to test.c correct?

Yes it is new; No, not intended for only `test.c`. Although it is only included in `test.c` now, future versions of the Espressif library _may_  support SHA interleaving (on SoC devices that support it). I envision this as being an optional feature. It will likely be valuable when testing performance between systems that have hardware-accelerated SHA interleaving enabled (a bit of overhead expected) vs no interleaving needed.

An example of _needing_ the interleave is the SSH signature. Other applications may not need nor want the interleaving overhead needed for hardware interleaving.

> I believe the interleave avoids the copy, update, final scenario?

Interesting question. No, not avoided. The interleave does perform the  _initialize_, update, and final (then copy to output). The interesting thing is that when for example one SHA256 hash is in progress, what happens to a concurrent SHA256 in progress at the same time in a different object.

There's _another_ type of copy that might be interesting to more exhaustively test as well: the copy of the `wc_Sha256` object to another instance. I could address that in a separate PR (or here?) is desired.

> Also needs documented.

Thanks for reminder. Added  https://github.com/wolfSSL/documentation/pull/120